language/stdlib/modules/DesignatedDealer.move

```diff
@@ -225,12 +280,34 @@ module DesignatedDealer {
                 dealer.window_inflow == old(dealer.window_inflow) + amount);
         ensures tier_index < len(old(dealer).tiers);
         ensures dealer.window_inflow <= old(dealer).tiers[tier_index];
+
+        include TieredMint<CoinType>;
+    }
+
+    spec schema TieredMint<CoinType> {
+        tc_account: signer;
+        dd_addr: address;
+        amount: u64;
+        tier_index: u64;
+        include Roles::AbortsIfNotTreasuryCompliance{account: tc_account};
+        aborts_if amount == 0 with Errors::INVALID_ARGUMENT;
+        include AbortsIfDealerNotExistAt;
+        aborts_if !exists<TierInfo<CoinType>>(dd_addr);
+        let tier_info = global<TierInfo<CoinType>>(dd_addr);
+        aborts_if tier_info.window_start + ONE_DAY > max_u64();
+        aborts_if LibraTimestamp::spec_now_microseconds() <= tier_info.window_start + ONE_DAY && tier_info.window_inflow + amount > max_u64();
+        aborts_if tier_index >= len(tier_info.tiers) with Errors::INVALID_ARGUMENT;
+        aborts_if LibraTimestamp::spec_now_microseconds() <= tier_info.window_start + ONE_DAY && tier_info.window_inflow + amount > tier_info.tiers[tier_index] with Errors::INVALID_ARGUMENT;
+        aborts_if LibraTimestamp::spec_now_microseconds() > tier_info.window_start + ONE_DAY && amount > tier_info.tiers[tier_index] with Errors::INVALID_ARGUMENT;
+        include LibraTimestamp::AbortsIfNoTime;
+        aborts_if !exists<Libra::MintCapability<CoinType>>(Signer::spec_address_of(tc_account));
+        include Libra::MintAbortsIf<CoinType>{value: amount};
     }
 
     public fun exists_at(dd_addr: address): bool {
         exists<Dealer>(dd_addr)
     }
-    spec schema AbortsIfNotExistAt{
+    spec schema AbortsIfDealerNotExistAt {
```

Comment:
Lets introduce an opaque spec also for exist_at? Makes things slightly better performing. This would do the somehow strange `ensures result == exists_at(...)`.

Reply:
Done.